apollo-federation/src/operation/optimize.rs

```diff
@@ -2986,66 +2986,6 @@ mod tests {
     /// applied directives
     ///
 
-    #[test]
-    #[should_panic(expected = "directive cannot be used on FRAGMENT_DEFINITION")]
-    // TODO: Investigate this restriction on query document in Rust version.
-    fn reuse_fragments_with_same_directive_on_the_fragment() {
```

Comment:
instead of deleting this test -> should we update it to use valid (custom) directive?

Reply:
Yeah on second thought you are right. I'm not sure what we even do if there's a custom directive on a fragment definition. There's a good chance we do something nonsensical and that warrants a test.